app/models/matching/engine.rb

```diff
@@ -71,7 +72,7 @@ def create_candidate_matches opportunity
       }
       match =
         client.candidate_matches.create(opportunity: opportunity, client: client, universe_state: universe_state)
-      match.activate! if client_priority == 1
+      match.activate! if client_priority == 1 && activate_priority_match
```

Comment:
I believe you don't need to pass `activate_priority_match` all the way down.  You should be able to get it from `opportunity.match_route.should_activate_match` or `match.match_route.should_activate_match`.  I believe that would minimize the changes here.